src/util/nfa.h

```diff
@@ -0,0 +1,168 @@
+/// \file
+/// \author Diffblue Ltd.
+///
+/// A simple NFA implementation
```

Comment:
Please could you add what it is intended for.  There have been other automaton implementations in CPROVER (for SATAbs for example) and I wouldn't like future people to get the wrong idea of what this is.

Reply:
Good point, will do